src/api/chainHead.md

```diff
@@ -14,21 +14,21 @@ This beginner guide shows how to use the `chainHead` functions in order to know
 
 6. The return value you've just saved is called the metadata, prefixed with its SCALE-compact-encoded length. You must decode and parse this metadata. How to do this is out of scope of this small guide. The metadata contains information about the layout of the storage of the chain. Inspect it to determine how to find the storage item you're looking for.
 
-7. In order to obtain a value in the storage, call `chainHead_unstable_storage` with `hash` equal to `finalizedBlockHash`, `key` the desired key, and `type` equal to `value`. If the JSON-RPC server instead sends back a `{"event": "operationInaccessible"}` notification, the value you're looking for is unfortunately inaccessible and you can either try again or give up. If the JSON-RPC server instead sends back a `{"event": "operationStorageItems"}` notification, you can find the desired value inside.
+7. In order to obtain a value in the storage, call `chainHead_unstable_storage` with `hash` equal to the _current finalized block_, `key` the desired key, and `type` equal to `value`. If the JSON-RPC server instead sends back a `{"event": "operationInaccessible"}` notification, the value you're looking for is unfortunately inaccessible and you can either try again or give up. If the JSON-RPC server instead sends back a `{"event": "operationStorageItems"}` notification, you can find the desired value inside.
 
 8. You are strongly encouraged to maintain [a `Set`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Set) of the blocks where the runtime changes. Whenever a `{"event": "newBlock"}` notification is received with `subscription` equal to your `followSubcriptionId`, and `newRuntime` is non-null, store the provided `blockHash` in this set.
 
-9. Whenever a `{"event": "finalized"}` notification is received with `subscription` equal to your `followSubcriptionId`, call `chainHead_unstable_unpin` once with your current `finalizedBlockHash`, once for each `prunedBlockHashes`, and once for each value in `finalizedBlockHashes` except for the last one. The last value in `finalizedBlockHashes` becomes your new `finalizedBlockHash`. If one or more entries of `finalizedBlockHashes` is found in your `Set` (see step 7), remove them from the set and jump to step 3 as the metadata has likely been modified. Otherwise, jump to step 7.
+9. Whenever a `{"event": "finalized"}` notification is received with `subscription` equal to your `followSubcriptionId`, call `chainHead_unstable_unpin` with _current finalized block_ hash, the `prunedBlockHashes`, and with each value in `finalizedBlockHashes` of the `finalized` event except for the last entry. The entry in `finalizedBlockHashes` becomes your new _current finalized block_. If one or more entries of `finalizedBlockHashes` are found in your `Set` (see step 7), remove them from the set and jump to step 3 as the metadata has likely been modified. Otherwise, jump to step 7.
```

Comment:
```suggestion
9. Whenever a `{"event": "finalized"}` notification is received with `subscription` equal to your `followSubcriptionId`, call `chainHead_unstable_unpin` with _current finalized block_ hash, the `prunedBlockHashes`, and with each value in `finalizedBlockHashes` of the `finalized` event except for the last entry. The last entry in `finalizedBlockHashes` becomes your new _current finalized block_. If one or more entries of `finalizedBlockHashes` are found in your `Set` (see step 7), remove them from the set and jump to step 3 as the metadata has likely been modified. Otherwise, jump to step 7.
```